docs/cody/capabilities/agentic-chat.mdx

```diff
@@ -0,0 +1,83 @@
+# Agentic Chat
+
+<p className="subtitle"> Learn about the agentic chat experience, an exclusive chat-based AI agent with enhanced capabilities.</p>
+
+<Callout type="info">Agentic chat is currently in the Experimental stage for Cody Pro and Enterprise and is supported on VS Code, JetBrains, Visual Studio editor extensions, and Web. Usage may be limited at this stage.</Callout>
+
+Cody's agentic chat experience is an AI agent that can evaluate context and fetch any additional context (OpenCtx, terminal, etc.) by providing enhanced, context-aware chat capabilities. It extends Cody's functionality by proactively understanding your coding environment and gathering relevant information based on your requests before responding. These features help you get noticeably higher-quality responses.
+
+This agentic chat experience aims to reduce the learning curve associated with traditional coding assistants by minimizing users' need to provide context manually. It achieves this through agentic context retrieval, where the AI autonomously gathers and analyzes context before generating a response.
+
+## Capabilities of agentic chat
+
+The agentic chat experience leverages several key capabilities, including:
+
+- **Proactive context gathering**: Automatically gathers relevant context from your codebase, project structure, and current task
+- **Agentic context reflection**: Review the gathered context to ensure it is comprehensive and relevant to your query
+- **Iterative context improvement**: Performs multiple review loops to refine the context and ensure a thorough understanding
+- **Enhanced response accuracy**: Leverages comprehensive context to provide more accurate and relevant responses, reducing the risk of hallucinations
+
+## Enable agentic chat
+
+Pro users can find the agentic chat option in the LLM selector drop-down. Enterprise customers must opt-in to access this agentic chat feature.
+
+![agentic chat interface](https://storage.googleapis.com/sourcegraph-assets/Docs/agenti%20-chat.jpg)
+
+### Getting agentic chat access for Enterprise customers
+
+Currently, agentic chat uses Claude 3.5 Haiku for the reflection steps as it provides a good balance between quality and latency. Your enterprise instance must have access to Claude 3.5 Haiku to use agentic chat.
+
+Additionally, enterprise customers need to upgrade their supported client (VS Code, JetBrains, and Visual Studio) to the latest version of the plugin by enabling the following feature flags on their Sourcegraph Instance:
+
+- `agentic-chat-experimental` to get access to the feature
+- `agentic-chat-cli-tool-experimental` to allow [terminal access](#terminal-commands)
+- Users will also need to manually enable access to the terminal commands on the client's `settings.json` file i.e., set `"cody.agentic.context.experimentalShell": true`
```

Comment:
```suggestion
```

Since all commands are now required manual approval before it runs, this setting has been removed